xtevent/_eventgenvars.ado

```diff
@@ -310,6 +332,64 @@ program define _eventgenvars, rclass
 		
 	}
 
+**************** find event-time limits based on observed data range ********
+
+	if "`w_type'"=="string" {
+		
+		qui xtset `panelvar' `timevar', noquery
+		qui sort `panelvar' `timevar', stable
+		
+		*create relative-to-event-time variable 
+		tempvar d1z ttreat ttreat2 rtime minrtime maxrtime
+		qui gen long `d1z'=d1.`zn2'
+		qui gen long `ttreat' = `timevar' if `d1z'!=0 & !missing(`d1z') & `touse'
+		qui by `panelvar' (`timevar'): egen long `ttreat2' = min(`ttreat') if `touse'
+		qui by `panelvar' (`timevar'): gen long `rtime' = `timevar' - `ttreat2' if `touse'
+		
+		if "`lwindow'"=="max" {
+			qui sum `rtime'
+			loc lwindow = r(min)
+			loc rwindow = r(max)
+		}
+		if "`lwindow'"=="balanced" {
+			qui by `panelvar' (`timevar'): egen long `minrtime' = min(`rtime') if `touse'
+			qui by `panelvar' (`timevar'): egen long `maxrtime' = max(`rtime') if !missing(`rtime') & `touse' 
+			qui sum `minrtime'
+			loc lwindow = r(max)
+			qui sum `maxrtime'
+			loc rwindow = r(min)
+		}
+		
+		*adjust for the endpoints 
+		loc lwindow = `lwindow' +1
+		loc rwindow = `rwindow' -1
+		
+		**** Error messages if calculated window limits are not valid  
+		if  (-`lwindow'<0 | `rwindow'<0) {

```

Comment:
@Constantino-Carreto-Romero The message displaying the calculated window limits below should appear even if there are no errors. It will be informative to know what are the window limits in all cases.
Also add an informative message on which window selection criteria was used.

Reply:
@jorpppp 
changed in https://github.com/JMSLab/xtevent/commit/267cef77ee4a32a1c26cce9d41e129d88cd45be0. 
I also changed an error message to be more informative.

Reply:
@Constantino-Carreto-Romero What does "around the event-time" mean? I think you mean the time when these first treated and last treated units are treated, right? If so change the message because we use "event-time" to refer to time to event in general.

Reply:
@jorpppp 
changed in https://github.com/JMSLab/xtevent/commit/154d09f0c66566bf6c41a6841031c0a4de895964